scenes/game_elements/characters/enemies/guard/components/guard.gd

```diff
@@ -504,22 +504,28 @@ func _set_alert_other_sound_stream(new_value: AudioStream) -> void:
 
 
 func _on_instant_detection_area_body_entered(body: Node2D) -> void:
+	if not body is Player:
+		return
 	state = State.ALERTED
-	player_detected.emit(body)
+	player_detected.emit(body as Player)
 
 
 func _on_detection_area_body_entered(body: Node2D) -> void:
-	_player = body
+	if not body is Player:
+		return
+	_player = body as Player
```

Comment:
But in this case I think the early return is clearer than indenting the whole function.